straxen/get_corrections.py

```diff
@@ -156,20 +156,43 @@ def _is_cmt_option(run_id, config):
 def get_cmt_options(context):
     """
     Function which loops over all plugin configs and returns dictionary
-    with option name as key and current settings as values.
+    with option name as key and a nested dict of CMT correction name and version as values.
 
     :param context: Context with registered plugins.
     """
+
     cmt_options = {}
+    runid_test_str = 'norunids!'
+
     for data_type, plugin in context._plugin_class_registry.items():
         for option_key, option in plugin.takes_config.items():
             if (option_key in context.config and
-                is_cmt_option(context.config[option_key])
-                ):
-                cmt_options[option_key] = context.config[option_key]
+                    is_cmt_option(context.config[option_key])):
+                opt = context.config[option_key]
             elif is_cmt_option(option.default):
-                cmt_options[option_key] = option.default
+                opt = option.default
+            else:
+                continue
+
+            # check if it's a URLConfig
+            if isinstance(opt, str) and 'cmt://' in opt:
+                before_cmt, cmt, after_cmt = opt.partition('cmt://')
+                p = context.get_single_plugin(runid_test_str, data_type)
+                url_config = p.takes_config[option_key]
+                url, url_kwargs = url_config.split_url_kwargs(after_cmt)
+                kwargs = {k: url_config.fetch_attribute(p, v) for k, v in url_kwargs.items()}
+                correction_name = url_config.dispatch(url, **kwargs)
+                version = kwargs.get('version', 'ONLINE')
+                cmt_options[option_key] = {'correction': correction_name,
+                                           'version': version,
+                                           'strax_option': opt,
+                                           }
 
+            else:
+                cmt_options[option_key] = {'correction': opt[0],
+                                           'version': opt[1],
+                                           'strax_option': opt,
+                                           }
```

Comment:
Just a matter of personal taste but i would probably have return a tuple of length 3 here that's compatible with the cmt structure. But I'm fine with an explicit dict too (obviously if i was building the format from scratch i would have done it the way you did, but its already a tuple so not sure its worth the risk of changing)

Reply:
I did that first but then that changes the lineage hash. For example running this function with `global_ONLINE` will overwrite the default options and make them tuples. I don't think this is what we want right?

Reply:
I dont follow, how would the structure of what you return here affect the lineage? its just the inputs to your `apply_cmt_version` function right? so why would the structure of the data you pass between the two functions have any affect on anything? its just the structure, a tuple vs a dict, the data is the same

Reply:
to be clear, im not talking about replacing the value of the config with a cmt compatible tuple. Im just talking about keeping the return value of this specific function consistent with what it was in the past. so e.g. `(cmt_name, cmt_version/url, True)`
Not super critical but would be backward compatible.

Reply:
Ah right we had this discussion before. It bothers me to have it return something like `(cmt_name, url, True)` for URLConfigs as this is basically nonsensical. We then would need to add another `if` statement to check what kind of config it is when we actualy do the `st.set_config`. I feel like either way we are breaking something, might as well make it something that can be more universal. 

If you insist though we can change. I was wrong before... 😉

Reply:
No no, I dont insist at all on the tuple. I think the chances of some current script somewhere using this function is negligible so probably wont break anything. But i would definitely like lean heavily towards not fetching the kwargs for some arbitrary run_id. If you want to return the kwargs then I would probably have this function accept run_id as an input so if anyone else decides to use this function at least its clear that the return value is run_id dependent.

Reply:
Thanks @jmosbacher. I dropped the kwargs part so now it only returns a dict of correction name and the strax option. I still prefer to return the dict -- it should be more future safe as we (hopefully) get rid of these tuple configs